scene/main/viewport.cpp

```diff
@@ -3848,7 +3848,67 @@ Viewport *Viewport::get_parent_viewport() const {
 
 void Viewport::set_embedding_subwindows(bool p_embed) {
 	ERR_THREAD_GUARD;
-	gui.embed_subwindows_hint = p_embed;
+	if (gui.embed_subwindows_hint == p_embed) {
+		return;
+	}
+
+	bool allow_change = true;
+
+	if (gui.embed_subwindows_hint) {
+		if (gui.sub_windows.size() > 0) {
+			// Prevent change when this viewport has embedded windows.
+			allow_change = false;
+		}
+	} else {
+		Viewport *vp = this;
+		while (true) {
+			if (!vp->get_parent()) {
+				break;
+			}
+			vp = vp->get_parent()->get_viewport();
+			if (!vp) {
+				// Necessary in cases when function is called on viewports that are not in the scene tree.
+				break;
+			}
+			if (vp->is_embedding_subwindows()) {
+				for (int i = 0; i < vp->gui.sub_windows.size(); i++) {
+					if (this->is_ancestor_of(vp->gui.sub_windows[i].window)) {
+						// Prevent change when this viewport has children windows, that are displayed in an ancestor viewport.
+						allow_change = false;
+						break;
+					}
+				}
```

Comment:
Hmm, when would that be possible? Does viewport embedding not follow the tree hierarchy and order?

Reply:
Take the following scene tree:
```
- Root Window (embed windows)
  - SubViewportContainer
    - SubViewport (don't embed windows)
      - Window
```
In this case the Window is embedded within the Root Window and not in the SubViewport.
So it is possible that Windows get embedded further up the scene tree and get not embedded in the nearest Viewport.

The above code was necessary to prevent changing `gui_embed_subwindows` of the SubViewport while the Window is displayed in embedded form in the Root Window.

Reply:
I see, though this is surprising. I would expect `Window` to only be affected by its closes Viewport parent. Was this discussed at any point, or was it like this already when you started looking into viewport related issues?

Reply:
As far as I can tell, this is the original behavior since Godot 4 Alpha.
I'm not aware, that it was discussed anywhere.